### PR TITLE
chore(eslint): add eslint-mdx package and fix eslint errors

### PR DESCRIPTION
**Pull request by imklau**
### What does it do?
Adds `eslint-mdx` package to the design system and `.mdx` file to the `lint` command.

### Why is it needed?
Without linting mdx files it's very common to have many unused variables imported or just other small bugs. Let's keep it clean ✨ 

### How to test it?

1. Run `yarn lint` & verify if there are any errors or warnings.
2. Make a mistake inside any `mdx` file, run `yarn lint` and verify if an error is shown.

### CODE REVIEW

File: packages/strapi-design-system/src/Radio/Radio.stories.mdx [hunk 1]
```diff
@@ -23,7 +24,7 @@ Radio buttons allow users to choose a single value from a range of options.
 - Radio buttons should be used only when it is not obvious enough that a choice is required.
 - Radio buttons should not be used in Table components.
 - Use Switch or Toggle component to make binary choices.
-[View source](https://github.com/strapi/design-system/tree/main/packages/strapi-design-system/src/Radio)
+  [View source](https://github.com/strapi/design-system/tree/main/packages/strapi-design-system/src/Radio)
```
[the remark this answers is not in the document]
Comment (HichamELBSI) [reply packages/strapi-design-system/src/Radio/Radio.stories.mdx:27]: The `View source` should be displayed on a new line 😅

File: packages/strapi-design-system/src/Switch/Switch.stories.mdx [hunk 2]
```diff
@@ -12,12 +11,13 @@ import { Switch } from './Switch';
...
 
 **Best practices**
+
 - Make sure to add state labels when using a Switch component. For inclusion matters, colors are mandatory but not enough to indicate a change of state.
 - The state labels should only have a single word and be clear.
 - A change of state should also be visible by the circle position.
 - One tap should be enough to change the state.
 - When in a form, prefer the usage of a Toggle component.
-[View source](https://github.com/strapi/design-system/tree/main/packages/strapi-design-system/src/Switch)
+  [View source](https://github.com/strapi/design-system/tree/main/packages/strapi-design-system/src/Switch)
```
Comment (HichamELBSI) [inline packages/strapi-design-system/src/Switch/Switch.stories.mdx:20]: Same here please 😅

File: packages/strapi-design-system/src/Radio/Radio.stories.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (imklau) [reply packages/strapi-design-system/src/Radio/Radio.stories.mdx]: @HichamELBSI done  ✅

File: packages/strapi-design-system/src/Switch/Switch.stories.mdx (hunk 2 above)
Comment (imklau) [reply packages/strapi-design-system/src/Switch/Switch.stories.mdx]: @HichamELBSI done  ✅

**Review by HichamELBSI — APPROVED** [review]
Awesome! Thank you @imklau

**Pull request merged**